docs/setup/independent/high-availability.md

```diff
@@ -0,0 +1,328 @@
+
+## Set up master LB
+
+1. The next step is to create a cloud Load Balancer for your master nodes. This solution will depend on your cloud provider. Examples are:
+
+- [AWS Elastic Load Balancer](https://aws.amazon.com/elasticloadbalancing/)
+- [GCE Load Balancing](https://cloud.google.com/compute/docs/load-balancing/)
+- [Azure](https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-overview)
+
+You will need to ensure that the load balancer routes to all three master VMs set up in the previous step **on port 6443**. If possible, use a smart load balancing algorithm like "least connections", and use health checks so unhealthy nodes can be removed from circulation. Most providers will provide these features.
+
+## Set up master VMs
+
+1. On each master VM, run the following:
+
+```
+cat >install.sh <<EOL
+apt-get update && apt-get install -y docker.io apt-transport-https
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+cat <<EOF >/etc/apt/sources.list.d/kubernetes.list
+deb http://apt.kubernetes.io/ kubernetes-xenial main
+EOF
+apt-get update
+apt-get install -y kubelet kubeadm kubectl
+EOL
+chmod +x ./install.sh
+./install.sh
+```
+
+This will install of the dependencies required for kubeadm to run.
+
+### acquire etcd certs
+
+1. Generate SSH keys for each of the master nodes by following the steps in the [create ssh access](#create-ssh-access) section. After doing this, each master will have an SSH key and will be permitted to SSH into `etcd0`.
+
+1. Next, run the following:
+
+```
+mkdir -p /etc/kubernetes/pki/etcd
+scp root@<etcd1-ip-address>:/root/certs/ca.pem /etc/kubernetes/pki/etcd
+scp root@<etcd1-ip-address>:/root/certs/client.pem /etc/kubernetes/pki/etcd
+scp root@<etcd1-ip-address>:/root/certs/client-key.pem /etc/kubernetes/pki/etcd
+```
+
+### use kubeadm
+
+1. The next step is to write the configuration file for kubeadm:
+
+```
+cat >config.yaml <<EOL
+apiVersion: kubeadm.k8s.io/v1alpha1
+kind: MasterConfiguration
+api:
+  advertiseAddress: <load-balancer-ip>
+etcd:
+  endpoints:
+  - https://<etcd0-ip>:2379
+  - https://<etcd1-ip>:2379
+  - https://<etcd2-ip>:2379
+  caFile: /etc/kubernetes/pki/etcd/ca.pem
+  certFile: /etc/kubernetes/pki/etcd/client.pem
+  keyFile: /etc/kubernetes/pki/etcd/client-key.pem
+networking:
+  podSubnet: 10.244.0.0/16
+apiServerCertSANs:
+- <load-balancer-ip>
+EOL
+```
+
+Ensure that `<load-balancer-ip>` is replaced with the public IPv4 of your load balancer. You will also need to replace `<etcd0-ip>`, `<etcd1-ip>` and `<etcd2-ip>` with the IP addresses of your three etcd nodes.
+
+**Note**: We are assuming that Flannel will be the CNI provider. If you want to use another provide like Weave, please read the [requirements section](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/#pod-network) and adjust the configuration file accordingly.
+
+1. When this is done, run kubeadm like so:
+
+```
+kubeadm init --config=config.yaml
```

Comment:
IMO, after `kubeadm init` on the first master1, it is necessary to copy Kubernetes CA certs on master 2 & 3 before running `kubeadm init` there. What do you think about?

Reply:
Good catch! Forgot to do this